airflow/models.py

```diff
@@ -711,126 +711,55 @@ def is_runnable(self, flag_upstream_failed=False):
         """
         return self.is_queueable(flag_upstream_failed) and not self.pool_full()
 
-    def are_dependents_done(self, main_session=None):
-        """
-        Checks whether the dependents of this task instance have all succeeded.
-        This is meant to be used by wait_for_downstream.
+    def are_dependencies_met(
+            self, main_session=None, flag_upstream_failed=False):
+
+        TR = TriggerRule
 
-        This is useful when you do not want to start processing the next
-        schedule of a task until the dependents are done. For instance,
-        if the task DROPs and recreates a table.
-        """
         session = main_session or settings.Session()
```

Comment:
Note to self, just in case we reuse this : no need to initialise the session (and close it), we're just passing it on